### PR TITLE
Fix non-bash output while detecting shell

### PR DESCRIPTION
**Pull request by ianchen-tw**
Make sure you have checked all steps below.

### Prerequisite
* [x] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [x] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.

### Description
- [x] Here are some details about my PR
`pyenv-init --detect-shell` will output non-bash script result while configuration files are not supported, since placeholder values are used for explanation.

### Tests
Test for the new function.

### CODE REVIEW

**Review by native-api — COMMENTED** [review]
> pyenv-init --detect-shell will output non-bash script result while configuration files are not supported, since placeholder values are used for explanation.

Maybe my English is not good enough, but I've no idea what this means or how it explains the purpose and necessity of the added function.

Perhaps if you could update the Pyenv-Installer PR to see how it is supposed to be used?

**Comment by native-api** [comment]
> Perhaps if you could update the Pyenv-Installer PR to show how it is supposed to be used?

Okay, you already did. Let's see then...

File: libexec/pyenv-init [hunk 1]
```diff
@@ -112,8 +114,15 @@ function detect_profile() {
     rc='~/.profile'
     ;;
   * )
-    profile='your shell'\''s login startup file'
-    rc='your shell'\''s interactive startup file'
+    # if no_config_place_holder is set,
+    # do not filled in any placehodler for config file
+    if [ -n "$no_config_place_holder" ]; then
```
Comment (native-api) [inline libexec/pyenv-init:119]: This should not be a global variable.

Either it should be an optional argument to the function, or an exit code should indicate whether shell detection was successful.

**Review by native-api — CHANGES_REQUESTED** [review]

File: libexec/pyenv-init (hunk 1 above)
Comment (ianchen-tw) [reply libexec/pyenv-init]: b502c9a9dffb6be593e189459e3cbacb8b15d960

**Pull request merged**